lib/cookie.js

```diff
@@ -1385,6 +1385,43 @@ CookieJar.prototype.clone = function(newStore, cb) {
   });
 };
 
+CAN_BE_SYNC.push('removeAllCookies');
+CookieJar.prototype.removeAllCookies = function(cb) {
+  var store = this.store;
+  if (store.removeAllCookies instanceof Function) {
```

Comment:
Just noticed that this is going to evaluate to `true` for all stores that inherit from the `Store` in `lib/store.js`. This will cause the callback to receive an error instead of using the fallback code. Probably needs to be changed to:

```js
if (store.removeAllCookies instanceof Function && store.removeAllCookies != Store.prototype.removeAllCookies) {
}
```

Reply:
This should get tested explicitly too: create a new `Store` class via inheritance and confirm that an error isn't passed to the callback, but instead the `getAllCookies`/`removeCookie` route is taken.